src/components/QrScan/QrScan.js

```diff
@@ -137,14 +110,42 @@ const QrScan = () => {
 
   // Call handleScan() from parent component when the QrScanner successfully reads data
   useEffect(() => {
+    const handleScan = (data) => {
+      if (healthCardPattern.test(data)) {
+        const match = data.match(healthCardPattern);
+        if (match.groups.multipleChunks) {
+          const chunkCount = +match.groups.chunkCount;
+          const currentChunkIndex = +match.groups.chunkIndex;
+          let tempScannedCodes = [...scannedCodes];
+          if (tempScannedCodes.length !== chunkCount) {
+            tempScannedCodes = new Array(chunkCount);
+            tempScannedCodes.fill(null, 0, chunkCount);
+          }
+
+          if (tempScannedCodes[currentChunkIndex - 1] === null) {
+            tempScannedCodes[currentChunkIndex - 1] = data;
+          }
+
+          if (tempScannedCodes.every((code) => code)) {
+            setQrCodes(tempScannedCodes);
+            history.push('/display-results');
+          }
+          setScannedCodes(tempScannedCodes);
```

Comment:
I think you could create a new useRef for the running list of scanned codes, rather than useState, and then remove the item from the useEffect dependency array. The above is messy but it seems to work.

Reply:
The `useRef` approach got rid of the console error without changing any other functionality. Thank you @mickohanlon23 @radamson @nkarip for your help with this!